### ISSUE TITLE
Windows node pool missing required nodeTaint

### ISSUE THREAD

**Issue by LucaPrete** [terraform-google-modules/cloud-foundation-fabric#374]
### TL;DR

When a Windows node pool is created, GKE implicitly adds a nodeTaint, so that linux pods don't get scheduled on the cluster.
When TF is applied again it finds the nodetaint and tries to remove it, causing the node pool to be re-created.

### Expected behavior

When a Windows node pool is created, the module needs to set a node taint, to be coherent with what GKE adds by itself, so that at the next apply, TF doesn't try to re-create it.

### Observed behavior

TF tries to re-create the node pool

### Terraform Configuration

```hcl
[after a GKE cluster is created...]

module "gke_2_nodepool" {
  source             = "git::https://github.com/terraform-google-modules/cloud-foundation-fabric.git//modules/gke-nodepool?ref=v8.0.0"
  name               = "np-${local.gke_win_machine_type}-002"
  project_id         = module.configs.project_ids.axis_dev_poc
  cluster_name       = module.gke_1.name
  location           = module.gke_1.location
  node_locations     = ["us-central1-a", "us-central1-f"]
  node_machine_type  = local.gke_win_machine_type
  initial_node_count = 2

  node_preemptible  = true
  max_pods_per_node = 100
  node_image_type   = "WINDOWS_SAC_CONTAINERD"
  autoscaling_config = {
    min_node_count = 2
    max_node_count = 4
  }

  management_config = {
    auto_repair  = true
    auto_upgrade = true
  }

  node_service_account_create = false
  node_service_account        = module.gke-nodepool-sa.email
  node_count                  = 2 #per-zone
  node_tags                   = ["np-${local.gke_win_machine_type}-002"]
}
```

### Terraform Version

```sh
v1.0.11
```

### Additional information

I'll send a PR.

This is the nodetaint that needs to be added:
node_taints       = ["node.kubernetes.io/os=windows:NoSchedule"]

**Comment by LucaPrete** [comment]
Fixed.
